### DISCUSSION TITLE
중급프로젝트 1주차 팀회고

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#429] [회고]
# 왜 회고를 해야하나요?
- **원활한 팀 프로젝트 진행을 위해, 회고가 좋은 도구가 되어줄거예요.** 
매주 팀 단위로 회고를 하며 어떤 점이 좋았고, 어떤 점이 아쉬웠고, 아쉬웠던 부분은 어떻게 개선해볼지 함께 이야기해보면 
그 다음주에는 더 나은 프로젝트를 진행할 수 있을거예요. 회고를 통해 함께 성장해보아요 🔥 
- **회고 시간만큼은 솔직한 생각, 마음을 터놓아보기! 😎** 
솔직한 회고를 통해 다음주는 더 즐겁게 프로젝트를 진행할 수 있을거예요.  

# 회고 방법
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/144666341/b1f1cc83-7d24-4156-910d-3fc9b6bb7555)
회고는 **“KPT 회고”** 방식으로 진행해볼거예요. 아래 방법대로 회고해주시면 됩니다. 
- 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
    - Keep: 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
    - Problem: 현재 우리팀이 가지고 있는 문제
    - Try: Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
- [KPT 회고 피그마 템플릿](https://www.figma.com/file/0NHIhlM9vhYZDrtZa7nJn8/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF?type=whiteboard&node-id=0%3A1&t=p6vXDVEjQntHKaNP-1)을 활용해서, 팀 회고를 진행하셔도 됩니다. 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요. 
- **이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.** 
(댓글에 예시 회고를 올려드릴테니, 참고해주세요. 피그마 템플릿으로 진행하셨다면 캡처본을 올려주셔도 돼요.)

# 참고 글
- [KPT 회고란 무엇인가?](https://brunch.co.kr/@jinha0802/35)
- [팀문화의 탄생](https://techblog.woowahan.com/2677/)

**Comment by bumsly** [comment]
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/65000254/cf71fc41-b0f4-4462-8710-a768a2c9d415)

**Reply by bumsly** [reply]
https://www.figma.com/file/a1fmZRmIgWfJ2vEygGnu5a/Untitled?type=design&node-id=0-1&mode=design&t=FKpFHVU3YBWM7Niy-0

**Comment by mizign** [comment]
## Keep
- (생각했던 것 보다) 공통 컴포넌트 다들 빠르게 만들고 있는 것 같다~
- 테일윈드 처음 써보는데 다들 적응하느라 애씀
- 브랜치 전략에 적응이 빠르다
- 브랜치 문제있을때 서로 잘도와줌 (화안냄 굿)
- 집단지성의 힘을 잘 발휘할 줄 앎 (밤늦게 디코 들어와줌)
- 대체적으로 팀워크가 좋음~

## Problem
- 팀 전체적으로 우유부단함; (뭔가 결정하는게 어려움)
- 더 생각나지 않음. (아직 본격적으로 페이지 시작을 안함..) 

## Try
- 이틀에 한 개 트러블 슈팅 한 거 공용 노션 페이지에 기록해두기. 자잘한 글감을 기록해두자.
- **문제** → 원인 → 해결 (짧게)

**Comment by Chyyni** [comment]
## 2-4팀

### Keep
- 프로젝트 시작 전 컨벤션을 구체적으로 정해 체계가 잡힌 느낌이라 좋습니다.
- 트러블 슈팅 문서 작성, 회의록 작성을 구체적으로 해서 좋습니다.
- 코어 타임에 서로 모르는 것에 대해 대화를 나누면서 진행해서 좋습니다.
- 이슈, 마일스톤을 통해 구체적인 목표를 명시적으로 확인할 수 있어서 좋습니다.

### Problem
- 스스로에 대한 걱정과 스트레스가 많은 분들이 많습니다.
- 지금도 코어 타임에 대화를 많이 하지만 더 많이 했으면 좋겠습니다.
- PR, branch flow에 대해 다들 명확히 숙지가 안된 상태입니다.

### Try
- 30분만 고민하고 물어보기와 같은 스스로의 규칙을 설정하는 방법도 좋을 것 같습니다.
- 멘토님께 PR, branch 다시 한번 물어봅시다.
- 코어 타임에 대화 더 많이 합시다.

**Comment by 9ummy9** [comment]
**7팀**
---

**Keep** 👌
   - 도움을 요청했을 때 친절하고 정확하게 답변을 해줘서 좋았다.
   - 팀원 모두 팀 미팅과 코어 타임을 잘 지키고 성실하다.
   - 근거와 함께 의견을 제시하고 서로의 의견을 잘 수용한다.
   - 컴포넌트 분배할 때 step별로 나눈게 좋았다.
   - 프로젝트 진행 상황이 노션에 잘 정리되어 보기 편했다.
   - 회의할 때 적극적으로 의견을 제시하고, 다른 팀원들의 의견도 적극적으로 반영해서 좋았다.

**Problem** 😢
   - merge할 때 대상 branch를 제대로 확인하고 PR을 작성한다.
   - tailwind를 사용하는데 러닝 커브가 있었다.
   - next.js에 익숙하지 않았다.
   - 초반에 코드 컨벤션이랑 PR 형식이 정해져 있지 않았었다.

**Try** 👍
   - 코드 리뷰를 좀 더 활발히 했으면 좋겠다.
   - 정해진 PR 양식을 잘 지키면 좋을 것 같다.
   - tailwind 공식 문서를 참고하며 익숙해지기 위해 노력한다.
   - twin.macro 사용 여부를 고민해보기로 했다.
   - 컴포넌트 선언 시 function을 사용하기로 했다.

<br>

![다운로드](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/f482d8c6-7664-4389-bb80-5e34a9ec8418)


7팀 파이팅!

**Comment by muyahho** [comment]
11팀  전우들의 회고록
![스크린샷 2024-01-28 140125](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/88885994/10fa4dce-275e-4da6-828c-e5447edac23d)

**Comment by designsoo** [comment]
# 8팀 회고록

<img width="888" alt="주간 리뷰-8팀" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/77719310/cdeda506-a5a1-4e7e-9101-a8a2ebda2354">

**Comment by rladngus133** [comment]
# 9팀 
## 📌 Keep
- 우현 : 초기셋팅 단계에서 생각히자 못했던 컨벤션 , 로직 등등 까지 팀원들이 고려해줘 좋았다.
- 하은 : 회의시간에 토론시에 팀원들이 서로를 존중하는 모습이 보여서 좋았다.
- 지원 : 다양한 기술들을 팀원들 모두 개방적으로 받아들여 좋았다.
- 오름 : 프로젝트에 관해 팀원들간의 건설적인 피드백이 자주 오고 간 것 같아 좋았다.
- 현기 : 맡은 부분을 일정에 맞춰 진행하는 부분이 좋았다.

## 📌 Problem
- 사소한 컨벤션을 간과하여 커밋=>풀 후 다시 고쳐야하는 상황이 발생하였다. (ex. 폴더이름 대소문자 , 단수형 복수형 등등)
- 새로운 기술을 사전지식없이 도입하여 약간의 어려움이 있었던것 같다. 

## 📌 Try
- 너무 작은 문제라 놓칠수있는 컨벤션도 꼼꼼하게 체크해보면 좋을것 같다.  
- 다 같이 새로운 기술 ( ex. axios 인터셉터 , react qurey 등등... )을 페어프로그래밍 등 협업을 통하여 구현해보려고한다.

9팀 파이팅 ~ 👊

**Comment by juncastle97** [comment]
# 3팀 회고록
### 😄Keep

- 다들 맡은 부분을 잘 해주고 있습니다.
- 피드백이 빠릅니다!
- 의견 반영을 잘 해줍니다.
- 결정하는 거에 있어서 시간 낭비를 하지 않고 빠르게 결정됩니다.

### 🥲Problem

- 계획이 자꾸 미뤄집니다.
- class명을 통일 하지 않았습니다.


### 😗Try

- 컨벤션 정한대로 잘 작성하기
- 더 부지런해지기
- 함께하는 시간 늘려보기
- 반대 의견 내보기(강요 아님)


3팀 파이팅!⭐️

**Comment by ynmkim** [comment]
## 10팀

### **Keep**

- 최소한의 공통 컴포넌트 작업 후, 페이지 도메인 별로 작업을 분담하여 책임과 자유가 있고, 그 안에 관련된 UI 구현, 기능 구현까지 한 흐름으로 경험해볼 수 있는 점이 좋습니다.
- 자유롭게 추가 의견을 제시할 수 있고, 기술적으로 개방적이고 유연한 분위기가 좋습니다.
- 새로운 기술들에 대해서 다양하게 이야기를 나누고 접해볼 수 있는 점이 좋습니다.
- 매일 진행 상황을 공유하고 추가 의견을 자유롭게 제시할 수 있는 점이 좋습니다.
- 어려움을 겪는 부분을 솔직하게 이야기할 수 있고, 해결할 수 있는 방법을 나눌 수 있는 점이 좋습니다.
- 실력이 부족하다고 느끼지만 압박하지 않고 서로 동기 부여 해주는 점이 좋습니다.

### Problem

- tailwind가 익숙하지 않은 팀원은 약간의 어려움이 있습니다. 하지만 장단점을 느껴 볼 수 있는 기회라고 생각합니다.
- 아직까지 큰 이슈없이 순조롭게 진행되고 있습니다.

### Try

- 의견을 조금 더 적극적으로 제안해도 괜찮아요!

**Comment by sung-eunji** [comment]
# 2팀

# 2024-01-29

##  Keep
- 프로젝트 시작 전 컨벤션을 구체적으로 정해 체계가 잡힌 느낌이라 좋아요.
- 회의록 작성을 구체적으로 해서 좋아요.
- 코어 타임에 서로 모르는 것에 대해 대화를 나누면서 진행해서 좋아요.
- 이슈를 통해 구체적인 목표를 명시적으로 확인할 수 있어서 좋아요.
- 서로 진행 상황에 대한 공유가 빨라서 좋아요.
- 팀원끼리 서로 돕고 도는 분위기가 좋아요.
- 팀원 모두 각자 맡은 바를 책임감 있게 잘 한다!
- 매일 회의록이 작성돼 협의된 내용을 까먹을 염려가 적다.

## Problem
- Git Convention이 잘 지켜지지 않는 거 같아요. 다시 한 번 점검해볼까요?
- PR Review 각자 한 번 씩 해보고 메세지 넣어주세요. (git CLI, Codespace 활용)
- 코드 리뷰를 안한다...
- 진척 상황에 대한 커뮤니케이션이 적다.

## Try
- 30분만 고민하고 물어보기와 같은 스스로의 규칙을 설정하는 방법도 좋을 것 같아요.
- 우리 여유를 가지고 만날 시간을 만들어보아요.
- 가끔 채팅방에서라도 같이 수다 떨어요.
- 코어 타임 정하기
- 코드 리뷰 하기

**Comment by eonpain** [comment]
# 5팀

## KEEP
-  내 파트가 아니더라도 개개인이 프로젝트 전반적인 내용들을 잘 인지하고 있다. 팀 내에서 서로 질문을 하면 답을 거의 찾아냄
-  다들 디스코드가 익숙하다. 위워크에 오지 않더라도 디스코드를 이용해 지속적으로 원활한 소통이 이루어짐
-  웬만하면 마일스톤을 지켜 그리 자세하지 않아도 계획을 확실히 지키는 편임

## PROBLEM
-  KEEP에 있는 사항이 모든 팀원들에게 적용되는 것은 아님. 소통의 문제와 계획을 지키는 것의 문제가 존재함
-  계획을 정할 때나 의견을 피력해야할 때 소극적임. 어떤 한 의견을 따라가는 경향이 있음

## TRY
-  아직 소통을 많이 하지 않는 팀원도 현재 상황, 닥친 문제, 시덥잖은 말, 심리적인 문제들 등등 자유롭게 소통하기
-  의견을 제시해야할 때, 어떤 선택을 해야할 때 적극적인 의견, 잘 알지 못하는 영역이라도 오히려 모르는 것을 질문하거나, 그 전에 스스로 구글링하는 습관 기르기

(프로젝트하는 깃헙 wiki에 1주차 피드백을 적어놨었는데 옮겨적지 못했네요 기간 내게 적지 못해서 죄송합니다)
